### PR TITLE
Fix some of the release pipeline installation test failures

### LINKED ISSUE #8014: When integrate the realm-swift that build with xcframework, there will be have many errors
### How frequently does the bug occur?

All the time

### Description

I have build the real-swift with xcframework and I integrate it in the xcode project. But there will have an error that is "Failed to build module  for importation due to the errors above; the textual interface may be broken by project issues or a compiler bug". This is a bug in Swift [SR-14195](https://github.com/apple/swift/issues/56573). Because the real-swift have the public method Realm that is conflict with Realm module. so how can I fix this bug? 

### Stacktrace & log output

```shell
SwiftRealm.framework/Modules/SwiftRealm.swiftmodule/arm64-apple-ios-simulator.private.swiftinterface:99:17: error: 'RLMApp' is not a member type of struct 'Realm'

SwiftRealm.framework/Modules/SwiftRealm.swiftmodule/arm64-apple-ios-simulator.private.swiftinterface:2462:23: note: 'Realm' declared here
@frozen public struct Realm {
                      ^
```

### Can you reproduce the bug?

Yes, always

### Reproduction Steps

step1: build the realmswift in xcframework

step2: integrate in xcode project

step3: import xcframework

### Version

10.32.0

### What SDK flavour are you using?

Local Database only

### Are you using encryption?

Yes, using encryption

### Platform OS and version(s)

iOS

### Build environment

Xcode version: 14
Dependency manager and version: OSX 12.6

### PR DESCRIPTION
**Pull request by tgoyne**
A bunch of these have been broken for a while, and unsurprisingly they stopped being just CI issues after a while. The things fixed so far:

1. With Xcode 14 Catalyst now uses the macos deployment target instead of the ios deployment target. We weren't setting it on the example project, so with Xcode 14.1 it was defaulting to macOS 13 and failing to run because our CI machines aren't running that version yet.
2. The deployment target on our installation examples weren't bumped to match Xcode 14's new minimums. Oddly this sometimes was a warning and sometimes was an error.
3. We were still testing the legacy fat frameworks for Carthage rather than XCFrameworks.
4. Carthage currently doesn't support building for watchOS or tvOS with Xcode 14 because it only builds schemes with bitcode enabled. Test these on Xcode 13 for now.
5. We added some more types which the Swift compiler doesn't handle correctly when generating swiftinterface files.
6. Carthage sometimes builds the Realm scheme before the Realm iOS Static scheme, which resulted in the app failing to link because it's not set up for a static framework. There doesn't appear to be a way to fix this since Carthage explicitly refuses to give any sort of control over what schemes it builds, so I made the test app work with both static and dynamic Realm.framework.
7. Our obj-c package was missing an arm64 slice for watchOS. The Swift package was fine because we ship copies of Realm.xcframework built with each Xcode version, but the obj-c package only contains one.

Not yet fixed:

1. xcodebuild sometimes hits the assertion failure "thread is already initializing this class!" inside the obj-c runtime. We need to retry when this happens.
2. Waiting for the dyld cache to update when resetting simulators sometimes hangs for a very long time. This might be something that should be done at CI build time.

A passing cocoa-pipeline job from this branch: https://ci.realm.io/blue/organizations/jenkins/cocoa-pipeline/detail/cocoa-pipeline/1859/pipeline

Fixes #8014.

### CODE REVIEW

**Review by leemaguire — COMMENTED** [review]
This LGTM once the Xcode cloud missing schemes issue is addressed.

**Comment by dianaafanador3** [comment]
> This LGTM once the Xcode cloud missing schemes issue is addressed.

Do you mean the failing Xcode Cloud jobs?, we will have this failing jobs until the migrate for Xcode Cloud project is merged.

**Comment by leemaguire** [comment]
> > This LGTM once the Xcode cloud missing schemes issue is addressed.
> 
> Do you mean the failing Xcode Cloud jobs?, we will have this failing jobs until the migrate for Xcode Cloud project is merged.

Cool, LGTM so!

**Review by leemaguire — APPROVED** [review]

File: CHANGELOG.md
Comment (dianaafanador3) [inline CHANGELOG.md:8]: nice

**Review by dianaafanador3 — APPROVED** [review]
Very nice PR!!! Just a small comment, which is not related to changes to the PR

File: examples/installation/build.sh
Comment (tgoyne) [reply examples/installation/build.sh:103]: Ah, good call. I'll make a PR on the docs repo to switch it to --use-xframeworks.

**Pull request merged**
